source/plugins/plugins.md

```diff
@@ -55,277 +55,52 @@ Extension points allow a plugin to override a key class in the Struts framework
-   </tr>
-   <tr>
-     <td>com.opensymphony.xwork2.conversion.impl.StringConverter</td>
-     <td>struts.converter.string</td>
-     <td>singleton</td>
-     <td>Converter used to convert any object to String and back</td>
-   </tr>
-   <tr>
-     <td>com.opensymphony.xwork2.conversion.ConversionPropertiesProcessor</td>
-     <td>struts.conversion.properties.processor</td>
-     <td>singleton</td>
-     <td>Process Properties to create converters</td>
-   </tr>
-   <tr>
-     <td>com.opensymphony.xwork2.conversion.ConversionPropertiesProcessor</td>
-     <td>struts.converter.file.processor</td>
-     <td>singleton</td>
-     <td>Process <class>-conversion.properties file create converters</class></td>
-   </tr>
-   <tr>
-     <td>com.opensymphony.xwork2.conversion.ConversionAnnotationProcessor</td>
-     <td>struts.converter.annotation.processor</td>
-     <td>singleton</td>
-     <td>Process TypeConversion annotation to create converters</td>
-   </tr>
-   <tr>
-     <td>com.opensymphony.xwork2.conversion.TypeConverterCreator</td>
-     <td>struts.converter.creator</td>
-     <td>singleton</td>
-     <td>Creates user converters</td>
-   </tr>
-   <tr>
-     <td>com.opensymphony.xwork2.conversion.TypeConverterHolder</td>
-     <td>struts.converter.holder</td>
-     <td>singleton</td>
-     <td>Holds user converters' instances</td>
-   </tr>
-   <tr>
-     <td>com.opensymphony.xwork2.util.TextParser</td>
-     <td>struts.expression.parser</td>
-     <td>singleton</td>
-     <td>Used to parse expressions like ${foo.bar} or %{bar.foo} but it is up tp the TextParser's
-         implementation what kind of opening char to use (#, $, %, etc)</td>
-   </tr>
-   <tr>
-     <td>com.opensymphony.xwork2.ExcludedPatternsChecker</td>
-     <td>struts.excludedPatterns.checker</td>
-     <td>request</td>
-     <td>Used across different interceptors to check if given string matches one of the excluded patterns</td>
-   </tr>
-   <tr>
-     <td>com.opensymphony.xwork2.AcceptedPatternsChecker</td>
-     <td>struts.acceptedPatterns.checker</td>
-     <td>request</td>
-     <td>Used across different interceptors to check if given string matches one of the accepted patterns</td>
-   </tr>
-   <tr>
-     <td>org.apache.struts2.util.ContentTypeMatcher</td>
-     <td>struts.contentTypeMatcher</td>
-     <td>singleton</td>
-     <td>Matches content type of uploaded files (since 2.3.22)</td>
-   </tr>
-   <tr>
-     <td>com.opensymphony.xwork2.LocalizedTextProvider</td>
-     <td>struts.localizedTextProvider</td>
-     <td>singleton</td>
-     <td>Provides access to resource bundles used to localise messages (since 2.5.11)</td>
-   </tr>
- </table>
-
-
-
-
-{% comment %}end snippet id=extensionPoints|lang=html|javadoc=true|org.apache.struts2.config.DefaultBeanSelectionProvider {% endcomment %}
+|Type|Property|Scope|Description|
+|----|--------|-----|-----------|
+|ObjectFactory|struts.objectFactory|singleton|Creates actions, results, and interceptors|
```

Comment:
All the types are gone

Reply:
I have added following tip at end of table

> Same packages org.apache.struts2 and com.opensymphony.xwork2 are refactored from Type.

I had to do this because currently the table is too wide and causes scroll on whole page even on my wide monitor :(

Reply:
Hm... I am not sure if this is a good idea, if table doesn't fit we should consider using a list or something else. Dropping packages from the type can confuse users (especially that we use two different packages).

Reply:
What's your idea about removing `Type` or `Scope` column to last column? if so, `Description` will be visible without scrolling and only `Type` or `Scope` column will need :)

Reply:
Looks good to me :)